amp-script-tutorial/finished_code/js/script.js

```diff
@@ -0,0 +1,54 @@
+console.log('Hello, amp-script!');
+
+const passwordBox = document.getElementById("passwordBox");
+const submitButton = document.getElementById("submitButton");
+
+const checkRegexes = {
+  lower: /[a-z]/,
+  upper: /[A-Z]/,
+  digit: /\d/,
+  special: /[^a-z\d]/i,
+  eight: /.{8}/
+};
+
+// This creates a closure that contains the functions we'll use and stores the HTML element
+function initCheckPassword(element) {
+  const checkPass = (el) => {
+    el.classList.remove("invalid");
+    el.classList.add("valid");
+  };
+
+  const checkFail = (el) => {
+    el.classList.remove("valid");
+    el.classList.add("invalid");
+  }
+
+/*
+ * Retrieve the contents of the password input. Check it against our regexes.
+ * If a rule passes, turn the on-screen text green. If it fails, turn it red.
+ * If all the rules pass, enable the Submit button!
+ */
+  const checkPassword = () => {
+    const password = element.value;
+    let failed = false;
+
+    for (const check in checkRegexes) {
+      let li = document.getElementById(check);
+
+      if (password.match(checkRegexes[check])) {
```

Comment:
A simpler way to do this would be:

```
const updateStatus = (el, hasPassed) => {
  el.classList.toggle("invalid", !hasPassed);
  el.classList.toggle("valid", hasPassed);
}

...


const hasPassed = password.match(checkRegexes[check])
updateStatus(li, hasPassed)
```

It's less explicit though, so I'm not sure if that's favorable over the current solution.

Reply:
I struggled with this as I wrote it. I often struggle with this exact thing.

I think I prefer your version, though.

Reply:
Actually, I'm seeing that this signature for `toggle` isn't supported in IE. (Which may be why I'd never heard of it.) So I think it would require a more complex construction for me to take this more compact approach.

Which is too bad, because I also wanted to update the `failed` flag using a Boolean operator. How often do you get to use one of those?